distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/SecurityPolicyConfigurator.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package ddf.catalog.test;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.codice.ddf.security.policy.context.ContextPolicy;
+import org.codice.ddf.security.policy.context.ContextPolicyManager;
+
+import ddf.common.test.AdminConfig;
+import ddf.common.test.ServiceManager;
+
+public class SecurityPolicyConfigurator implements AdminConfig.ConfigWaitable {
+    private static final String SYMBOLIC_NAME = "security-policy-context";
+
+    private static final String FACTORY_PID = "org.codice.ddf.security.policy.context.impl.PolicyManager";
+
+    private static final String[] AUTH_TYPES = {"/=SAML|%s", "/admin=SAML|basic",
+            "/jolokia=SAML|basic", "/system=SAML|basic", "/solr=SAML|PKI|basic"};
+
+    private static final Lock LOCK = new ReentrantLock();
+
+    private static SecurityPolicyConfigurator anonConfig;
+
+    private static SecurityPolicyConfigurator basicConfig;
+
+    private final ServiceManager serviceManager;
+
+    private final String serviceRoot;
+
+    private final String policyType;
+
+    private final ContextPolicyManager ctxPolicyMgr;
+
+    private SecurityPolicyConfigurator(ServiceManager serviceManager, String serviceRoot,
+            String policyType) {
+        this.serviceManager = serviceManager;
+        this.serviceRoot = serviceRoot;
+        this.policyType = policyType;
+        ctxPolicyMgr = serviceManager.getService(ContextPolicyManager.class);
+    }
+
+    public static void configureRestForAnonymous(AdminConfig adminConfig,
+            ServiceManager serviceManager, String serviceRoot)
+            throws IOException, InterruptedException {
+        LOCK.lock();
```

Comment:
Why are you using a locking singleton pattern? This seems overly rigid to me, and also violates its own method contract. I think this needs to either return a new instance, update the singleton, or (and I don't like this option) throw an exception if it is recalled with a different serviceRoot.

I'd also like to see these two convenience methods pass through to a more generic method, both for future extensibility and to reduce code duplication.

It also looks like two of the three method parameters are just getters on AbstractIntegrationTest, so why not push that down in here, instead of requiring the using class to know about it? I think this class either isn't hiding enough or isn't generic enough. Or both. ;)

Also, why not update the config inside the lock as well? Of all things, that's the one you most want to prevent racing on, right?